lib/puppet/data_providers.rb

```diff
@@ -14,21 +14,21 @@ def self.assert_loaded
     @loaded = true
   end
 
-  def self.lookup_in_environment(name, scope)
+  def self.lookup_in_environment(name, scope, merge)
     assert_loaded()
     adapter = Puppet::DataProviders::DataAdapter.adapt(Puppet.lookup(:current_environment))
```

Comment:
I suspect there is a bug on line 6 above where the class `DataAdapters` (plural) is declared as an empty class to make that namespace available. I wonder if that namespace is ever used.